ecs-agent/daemonimages/csidriver/driver/mount.go

```diff
@@ -14,18 +14,39 @@
 package driver
 
 import (
-	"github.com/aws/amazon-ecs-agent/ecs-agent/daemonimages/csidriver/mounter"
+	"os"
+	"path/filepath"
 
+	"github.com/aws/amazon-ecs-agent/ecs-agent/daemonimages/csidriver/mounter"
 	mountutils "k8s.io/mount-utils"
 )
 
-// Mounter defines an interface for many volume related options. As of now, only
-// 'PathExists' is added to determine if a file path exists on the node.
+// Mounter is the interface implemented by NodeMounter.
+// A mix & match of functions defined in upstream libraries. (FormatAndMount
+// from struct SafeFormatAndMount, PathExists from an old edition of
+// mount.Interface). Define it explicitly so that it can be mocked and to
+// insulate from oft-changing upstream interfaces/structs
 type Mounter interface {
+	mountutils.Interface
+
+	FormatAndMountSensitiveWithFormatOptions(source string, target string, fstype string, options []string, sensitiveOptions []string, formatOptions []string) error
+	IsCorruptedMnt(err error) bool
+	GetDeviceNameFromMount(mountPath string) (string, int, error)
+	MakeFile(path string) error
+	MakeDir(path string) error
 	PathExists(path string) (bool, error)
+	NeedResize(devicePath string, deviceMountPath string) (bool, error)
+	Unpublish(path string) error
+	Unstage(path string) error
+	NewResizeFs() (Resizefs, error)
```

Comment:
I don't think it's very common that type A's interface requires another type B's initializer to be implemented. Also since mountutils's methods are included in the interface, can't we just use the `NewResizeFs()` there when we create a `ResizeFs`? Why do we need this wrapper that couple these two types?